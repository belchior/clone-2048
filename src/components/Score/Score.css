@import '../../static/vars.css';

@media all {
  .Score {
    display: flex;
    justify-content: space-between;
    font-family: 'Montserrat Alternates';
  }
  .Score .current_score,
  .Score .best_score {
    position: relative;
    display: flex;
    justify-content: flex-end;
    align-items: center;
    padding: 1rem 1rem 0.5rem 1rem;
    color: #ffffff;
    flex: 1;

  }
  .Score .current_score::before,
  .Score .best_score::before {
    position: absolute;
    top: 1rem;
    left: 0.5rem;
    font-size: 0.7rem;
    transform: translate3d(0, -0.8rem, 0);
  }
  .Score .current_score {
    background-color: var(--green);
  }
  .Score .current_score::before {
    content: 'score';
  }
  .Score .best_score {
    background-color: var(--yellow);
  }
  .Score .best_score::before {
    content: 'best score';
  }
}
