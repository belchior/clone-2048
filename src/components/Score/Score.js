import React from 'react';
import PropTypes from 'prop-types';

import './Score.css';

export const Score = ({current = 0, best = 0}) => (
  <div className="Score">
    <span className="current_score" title="The current score">{current}</span>
    <span className="best_score" title="The best score">{best}</span>
  </div>
);

Score.displayName = 'Score';
Score.propTypes = {
  current: PropTypes.number,
  best: PropTypes.number,
};
