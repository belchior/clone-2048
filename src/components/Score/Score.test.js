import React from 'react';
import { Score } from './Score';
import renderer from 'react-test-renderer';

it('Score should render without crashing', () => {
  let tree = renderer.create(<Score current={1000} best={1234}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Score should assume 0 as default value to current and best', () => {
  let tree = renderer.create(<Score/>).toJSON();
  expect(tree).toMatchSnapshot();
});
