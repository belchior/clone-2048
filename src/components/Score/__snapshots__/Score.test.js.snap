// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score should assume 0 as default value to current and best 1`] = `
<div
  className="Score"
>
  <span
    className="current_score"
    title="The current score"
  >
    0
  </span>
  <span
    className="best_score"
    title="The best score"
  >
    0
  </span>
</div>
`;

exports[`Score should render without crashing 1`] = `
<div
  className="Score"
>
  <span
    className="current_score"
    title="The current score"
  >
    1000
  </span>
  <span
    className="best_score"
    title="The best score"
  >
    1234
  </span>
</div>
`;
